src/main/scala/viper/gobra/translator/implementations/translator/StatementsImpl.scala

```diff
@@ -37,6 +39,26 @@ class StatementsImpl extends Statements {
     def goA(a: in.Assertion): CodeWriter[vpr.Exp] = ctx.ass.translate(a)(ctx)
     def goE(e: in.Expr): CodeWriter[vpr.Exp] = ctx.expr.translate(e)(ctx)
 
+    /**
+      * Translates a go call to a function or method with pre-condition `pre` which is parameterized by
+      * formal parameters `formalParams` and is instantiated with `args`
+      */
+    def translateGoCall(pre: Vector[in.Assertion],
+                        formalParams: Vector[in.Parameter.In],
+                        args: Vector[in.Expr]): Writer[vpr.Stmt] = {
+      Violation.violation(args.length == formalParams.length, "number of passed arguments must match number of expected arguments")
+      val errorT = (x: Source.Verifier.Info, _: ErrorReason) => PreconditionError(x) dueTo GoCallPreconditionError(x)
+      for {
+        vArgss <- sequence(args map goE)
+        funcArgs <- sequence(formalParams map goE)
+        substitutions = (funcArgs zip vArgss).toMap
+        preCond <- sequence(pre map goA)
+        preCondInstance = preCond.map{ _.replace(substitutions) }
+        and = vu.bigAnd(preCondInstance)(pos, info, errT)
+        _ <- assert(and, errorT)
+      } yield vpr.Exhale(and)(pos, info, errT)
```

Comment:
Here, you are first asserting and then exhaling the conjunct. The assertion is superfluous. You should add the error trasnformer directly to the exhale statement.